### DISCUSSION TITLE
Error with Postal.

### DISCUSSION THREAD

**Discussion by Alauraa** [postalserver/postal#1781] [Help with using Postal]
Im getting this error with postal, reverse proxying with nginx.

> The change you wanted was rejected.
Maybe you tried to change something you didn't have access to.

>If you are the application owner check the logs for more information.

OS System Info:
Arch: AARCH64
OS: Ubuntu 20.04
Ram: 24 GB's
CPU's: 4 OCI Cores

Not sure how to fix this, as the web side doesn't give me any info as to why this is happening.

**Comment by willpower232** [comment]
You'll need to have a look at the logs inside the container to see the error message. Are you seeing that all the time or only after you do something specific?

**Comment by Alauraa** [comment]
All the time, anytime i login i get the error, and i don't see anything in the logs that fixes it.

**Reply by willpower232** [reply]
If you run `postal bash web` then you'll be in the container and can have a look at /opt/postal/logs I think as the docker logs won't be what you need in this case.

**Reply by Alauraa** [reply]
I do have a Q tho for postal, it can run on ARM right? AARCH64 and all that?

**Reply by willpower232** [reply]
sorry I should have noticed that, I think some have got it working, see #1557 #1601 #1580, but they had to make their own docker image

**Reply by Alauraa** [reply]
So thank you, #1557 worked perfectly, hopefully one day postal can incorporate the install instructions for ARM into its documentation, would be very handy as well as nice for people who run ARM and then have documentation for those who run on AMD64 and what not.

**Reply by Alauraa** [reply]
I, [2021-12-24T10:29:59.196143 #9]  INFO -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] Started POST "/login" for 127.0.0.1 at 2021-12-24 10:29:59 +0000
I, [2021-12-24T10:29:59.196871 #9]  INFO -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] Processing by SessionsController#create as HTML
I, [2021-12-24T10:29:59.196971 #9]  INFO -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1]   Parameters: {"utf8"=>"✓", "authenticity_token"=>"OiVZZV0fzL1MhwsTctl8Oj9vLedWqfXHzDh561RnGmmLWykPhpba8WXHWMIvG9C/3gs2Tu$
W, [2021-12-24T10:29:59.197519 #9]  WARN -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] HTTP Origin header (https://example.domain.com) didn't match request.base_url (http://localhost:5000)
I, [2021-12-24T10:29:59.197838 #9]  INFO -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] Completed 422 Unprocessable Entity in 1ms (ActiveRecord: 0.0ms)
F, [2021-12-24T10:29:59.198413 #9] FATAL -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1]
F, [2021-12-24T10:29:59.198451 #9] FATAL -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] ActionController::InvalidAuthenticityToken (ActionController::InvalidAuthenticityToken):
F, [2021-12-24T10:29:59.198472 #9] FATAL -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1]
F, [2021-12-24T10:29:59.198496 #9] FATAL -- : [fdc3e6f7-8775-4ade-92da-cb5c74ddcbf1] lib/postal/tracking_middleware.rb:12:in `call'

This is the error i get by the way. As a note, i am running this under nginx, not caddy, so i don't know if i need to add any HTTP headers or what not to the config.

**Reply by willpower232** [reply]
> HTTP Origin header (https://example.domain.com) didn't match request.base_url (http://localhost:5000)

What does your nginx configuration look like?

**Reply by Alauraa** [reply]
https://www.toptal.com/developers/hastebin/yolulekejo.nginx This is what it looks like.

**Reply by willpower232** [reply]
Interesting, does it work if you use http instead of https? The https seems to be redirecting me for some reason.

**Reply by Alauraa** [reply]
It does the same thing as the https does " The change you wanted was rejected.
Maybe you tried to change something you didn't have access to.

If you are the application owner check the logs for more information."

**Reply by willpower232** [reply]
Do you have HAProxy involved somewhere perchance? It looks like there were some specific settings required to get passed this error message https://github.com/postalserver/postal/discussions/1785

**Reply by Alauraa** [reply]
Nope, purely nginx.

**Comment by Ascar699** [comment]
Hello, Merry Christmas!

I have a similar scenario right now - I put Postal behind HAProxy running on pFsense.  This is how things look for me:

1. I can access Postal by private IP address on the server through VPN, so if I enter http://192.168.2.104:5000 then the login page opens and I can log into Postal.
2. Until I published Postal through HAProxy I could do the same thing by using public IP address http://192.154.xxx.xxx:5000.  Was able to login also.
3. After publishing through HAProxy on pFsense my access pattern has changed to https://postal.mysite.com and the login page loads and renders properly with valid SSL certificate detected.  When I enter user name and password I get the same error 

![image](https://user-images.githubusercontent.com/28905723/147353179-83071231-945e-49e2-ab86-e070dff96891.png)

**Reply by Ascar699** [reply]
Small update:

Today I forwarded all ports on the firewall for the public IP address tied to my Postal A record to the VM with private IP address where Postal is installed - this means I removed my Postal instance from HAProxy and exposed my Postal VM to the internet.

After that I deployed Postal with Caddy and it works, i.e. I can log into Postal when opening the management site on standard SSL port and Caddy is proxying 443 to 5000 on the VM.

Will do some more testing now and post an update here.
